api/client/ps/custom.go

```diff
@@ -65,7 +65,9 @@ func (c *containerContext) Image() string {
 		return "<no image>"
 	}
 	if c.trunc {
-		return stringutils.Truncate(c.c.Image, 12)
+		if len(c.c.Image) == 64 {
```

Comment:
`==64` here is because I modified `daemon/list.go` (see diff below) will always return the full image ID which length is `64`.  if the length of the `c.c.Image` is 64, it means it is a image ID, so truncate it by default, if length is not 64, means it's not a ID, it's a name, so don't truncate it.